src/validation.cpp

```diff
@@ -1284,8 +1281,10 @@ void InitScriptExecutionCache() {
 }
 
 /**
- * Check whether all inputs of this transaction are valid (no double spends, scripts & sigs, amounts)
- * This does not modify the UTXO set.
+ * Check whether all of this transaction's input scripts succeed.
+ *
+ * This involves ECDSA signature checks so can be computationally intensive. This function should
+ * only be called after the basic tx amount tests in CheckTxInputs passed.
```

Comment:
This seems to imply that when the basic tx amount tests pass, the amount fields in the tx are correct. However, you can only know that the amount fields are correct when all subsequent check (including signature checks have been run). I'd rather say something like "all cheap sanity checks passed", to be not overly specific.

Reply:
agree. Fixed